src/Logary.Prometheus/Exporter.fs

```diff
@@ -0,0 +1,171 @@
+﻿namespace Logary.Prometheus
+open System
+open System.Text
+open System.Text.RegularExpressions
+open Logary.Metric
+open Suave
+open Suave.Filters
+open Suave.Operators
+open Suave.Writers
+open Hopac
+
+
+type ExporterConf =
+  {
+    metricNameTransformer: string -> string
+    metricRegistry: MetricRegistry
+    webConfig: SuaveConfig
+    urlPath: string
+  }
+
+module ExporterConf =
+  let validNameRegex = Regex("^[a-zA-Z_][a-zA-Z0-9_]*$", RegexOptions.Compiled)
+  let reservedNameRegex =  Regex("^__.*$", RegexOptions.Compiled)
+  let startCharValidRegex =  Regex("^[a-zA-Z_].*$", RegexOptions.Compiled)
+  let notValidAfterPartRegex =  Regex("[^a-zA-Z0-9_]", RegexOptions.Compiled)
+
+  let defaultMetricNameTrans metricName =
+    let mutable nameResult = metricName
+    if String.IsNullOrEmpty nameResult then failwith "name can't be null"
+    if reservedNameRegex.IsMatch nameResult then
+      nameResult <- "metric_" + nameResult.Substring 2
+    if not <| startCharValidRegex.IsMatch nameResult then
+      nameResult <- "metric_" + nameResult
+    notValidAfterPartRegex.Replace(nameResult, "_")
+
+  let nameValidator name =
+    if not <| validNameRegex.IsMatch name then
+      failwithf "name (%s) should match regex %s" name (validNameRegex.ToString())
+    elif reservedNameRegex.IsMatch name then
+      failwithf "(%s) is invalid, names beginning with __ are reserved for internal use." name
+    else name
+
+  let create (urlPath: string) (registry: MetricRegistry) =
+    {
+      metricNameTransformer = defaultMetricNameTrans
+      metricRegistry = registry
+      webConfig = defaultConfig
+      urlPath =  urlPath
+    }
+
+  let webConfig webConfig conf =
+    { conf with webConfig = webConfig }
+
+
+module Exporter =
+  open System.Threading
+  open System.Globalization
+  open Logary.Internals.Chiron.Formatting
+  open System.IO
+
+  /// HELP lines may contain any sequence of UTF-8 characters (after the metric name),
+  /// but the backslash and the line feed characters have to be escaped as \\ and \n, respectively
+  let escapeHelp (help: string) =
+    help.Replace("\\", @"\\").Replace("\n",@"\n")
```

Comment:
I'm not going to argue here at all, but we should consider using `netcoreapp2.2` in the future, and defaulting to the non-allocating Span-based API:s.

Reply:
Added a issue for tracking